packages/cloud_functions/cloud_functions/ios/cloud_functions/Sources/cloud_functions/FirebaseFunctionsPlugin.swift

```diff
@@ -0,0 +1,179 @@
+      binaryMessenger: registrar.messenger()
+    )
+    let instance = FirebaseFunctionsPlugin()
+    registrar.addMethodCallDelegate(instance, channel: channel)
+  }
+
+  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
+    guard call.method == "FirebaseFunctions#call" else {
+      result(FlutterMethodNotImplemented)
+      return
+    }
+
+    guard let arguments = call.arguments as? [String: Any] else {
+      result(FlutterError(code: "invalid_arguments",
+                          message: "Invalid arguments",
+                          details: nil))
+      return
+    }
+
+    httpsFunctionCall(arguments: arguments) { success, error in
+      if let error {
+        result(error)
+      } else {
+        result(success)
+      }
+    }
+  }
+
+  private func httpsFunctionCall(arguments: [String: Any],
+                                 completion: @escaping (Any?, FlutterError?) -> Void) {
+    let appName = arguments["appName"] as? String ?? ""
+    let functionName = arguments["functionName"] as? String
+    let functionUri = arguments["functionUri"] as? String
+    let origin = arguments["origin"] as? String
+    let region = arguments["region"] as? String
+    let timeout = arguments["timeout"] as? Double
+    let parameters = arguments["parameters"]
+    let limitedUseAppCheckToken = arguments["limitedUseAppCheckToken"] as? Bool ?? false
+
+    let app = FLTFirebasePlugin.firebaseAppNamed(appName)!
+
+    let functions = Functions.functions(app: app, region: region ?? "")
+
+    if let origin, !origin.isEmpty {
+      if let url = URL(string: origin) {
+        functions.useEmulator(withHost: url.host ?? "", port: url.port ?? 5001)
+      }
+    }
+
+    let options = HTTPSCallableOptions(requireLimitedUseAppCheckTokens: limitedUseAppCheckToken)
+
+    let function: HTTPSCallable
+
+    do {
+      if let functionName, !functionName.isEmpty {
+        function = functions.httpsCallable(functionName, options: options)
+      } else if let functionUri, !functionUri.isEmpty,
+                let url = URL(string: functionUri) {
+        function = functions.httpsCallable(url, options: options)
+      } else {
+        throw NSError(
+          domain: "com.firebase.functions",
+          code: -1,
+          userInfo: [
+            NSLocalizedDescriptionKey: "Either functionName or functionUri must be set",
+          ]
+        )
+      }
+
+      // Set timeout if provided
+      if let timeout {
+        function.timeoutInterval = timeout / 1000
+      }
+
+      function.call(parameters) { result, error in
+        if let error {
+          let flutterError = self.createFlutterError(from: error)
+          completion(nil, flutterError)
+        } else {
+          completion(result?.data, nil)
+        }
+      }
+    } catch {
+      completion(nil, FlutterError(
+        code: "illegal_argument",
+        message: error.localizedDescription,
+        details: nil
+      ))
+    }
```

Comment:
Is this do/catch necessary? It wasn't in previous implementation and we catch or pass exception back via completion handler